src/pypromice/process/join_l3.py

```diff
@@ -0,0 +1,282 @@
+    )
+    
+    station_dict = build_station_dict(args.config_folder)
+
+    l3m = xr.Dataset()
+    for stid in  station_dict[args.site]:
+        logger.info(stid)
+        
+        is_promice = False
+        is_gcnet = False
+        filepath = os.path.join(args.folder_l3, stid, stid+'_hour.nc')
+        if os.path.isfile(filepath):
+            is_promice = True
+        else:
+            filepath = os.path.join(args.folder_gcnet, stid+'.csv')
+            if os.path.isfile(filepath):
+                is_gcnet = True
+        if not is_promice and not is_gcnet:            
+            logger.info(stid+' not found either in '+args.folder_l3+' or '+args.folder_gcnet)
+            continue
+
+        l3, _ = loadArr(filepath)
+        
+        if is_gcnet:
+            l3 = gcnet_postprocessing(l3)
+        
+        # lat, lon and alt should be just variables, not coordinates
+        if 'lat' in l3.keys():
+            l3 = l3.reset_coords(['lat', 'lon', 'alt'])
+
+        if len(l3m)==0:
+            # saving attributes of station under an attribute called $stid
+            l3m.attrs[stid] = l3.attrs.copy()
+            # then stripping attributes
+            attrs_list = list(l3.attrs.keys())
+            for k in attrs_list:
+                del l3.attrs[k]
+                
+            l3m = l3.copy()
+        else:
+            # if l3 (older data) is missing variables compared to l3m (newer data)
+            # , then we fill them with nan
+            for v in l3m.data_vars:
+                if  v not in l3.data_vars:
+                    l3[v] = l3.t_u*np.nan
+                    
+            # if l3 (older data) has variables that does not have l3m (newer data)
+            # then they are removed from l3
+            list_dropped = []
+            for v in l3.data_vars:
+                if v not in l3m.data_vars:
+                    if v != 'z_stake':
+                       list_dropped.append(v)
+                       l3 = l3.drop(v)
+                    else:
+                       l3m[v] = ('time', l3m.t_u.data*np.nan)
+            logger.info('Unused variables in older dataset: '+' '.join(list_dropped))
+                        
+            # saving attributes of station under an attribute called $stid
+            st_attrs = l3m.attrs.get('stations_attributes', {})
+            st_attrs[stid] = l3.attrs.copy()
+            l3m.attrs["stations_attributes"] = st_attrs
+
+            # then stripping attributes
+            attrs_list = list(l3.attrs.keys())
+            for k in attrs_list:
+                del l3.attrs[k]
+            
+            l3m.attrs['stations_attributes'][stid]['first_timestamp'] = l3.time.isel(time=0).dt.strftime( date_format='%Y-%m-%d %H:%M:%S').item()
+            l3m.attrs['stations_attributes'][stid]['last_timestamp'] = l3m.time.isel(time=0).dt.strftime( date_format='%Y-%m-%d %H:%M:%S').item()
+
+            # merging by time block
+            l3m = xr.concat((l3.sel(
```

Comment:
Consider making a list in of site datasets in this loop ant the invoke xr.concat afterwards.

An approach the the merge could be something like:

* Add all station datasets to a list
* Slice the datasets based on their precedence.
* Concatenate the datasets using `xr.concat`
* Filter the variables based on the L3 variables on `variables.csv`

Reply:
I agree that, when working with non-overlapping slices, concatenation can be done at the end.

However, I'd like to keep it like this because more advanced combination should be developed in the future: pressure could be merged with "combine_first" while height-dependent variables like t, rh, wspd need to be filled at once on time slices.